fluentinterface/src/main/java/com/iluwatar/fluentinterface/fluentiterable/simple/SimpleFluentIterable.java

```diff
@@ -0,0 +1,194 @@
+package com.iluwatar.fluentinterface.fluentiterable.simple;
+
+import com.iluwatar.fluentinterface.fluentiterable.FluentIterable;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * This is a simple implementation of the FluentIterable interface. It evaluates
+ * all chained operations eagerly.
+ * @param <TYPE> the type of the objects the iteration is about
+ */
+public class SimpleFluentIterable<TYPE> implements FluentIterable<TYPE> {
+
+    private final Iterable<TYPE> iterable;
+
+    /**
+     * This constructor creates a copy of a given iterable's contents.
+     * @param iterable the iterable this interface copies to work on.
+     */
+    protected SimpleFluentIterable(Iterable<TYPE> iterable) {
```

Comment:
I added a copyFactoryMethod for the simple implementation. I wait before I push the commit because one thing has to be discussed first: I'm not sure if I implemented the lazy version correct/as you expected: It doesn't touch the incoming collection, because on termination a new iterator of the incoming iteratable is requested, used for iteration but no element is ever removed. Otherwise, one would have to handle ConcurrentModificationException or OperationNotSupportedException and I'm currently not quite sure if this is easy doable.